tests/phpunit/includes/testcase-trait.php

```diff
@@ -72,20 +86,55 @@ public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) { // p
 		remove_action( 'admin_print_styles', 'print_emoji_styles' );
 
 		self::filter_doing_it_wrong_trigger_error();
+
+		/*
+		 * Even though `$factory` should always be an instance of `PLL_UnitTest_Factory`,
+		 * it allows us to safely type hint `self::pllSetUpBeforeClass()`.
+		 */
+		if ( $factory instanceof PLL_UnitTest_Factory ) {
+			static::pllSetUpBeforeClass( $factory );
+		}
```

Comment:
I wonder it would be ok not to check the instance:
My idea is that if for some obscure reason the factory is of the wrong type, the type error will tell us immediately what's wrong (instead of trying to understand why the method is not executed as expected.

Reply:
As said, to avoid other testcases (Ajax, Canonical) to get a fatal, we have to keep it that way. I've updated the comment though (see d355d880).